packages/wallet/contract/src/smartWallet.js

```diff
@@ -4,87 +4,264 @@ import {
     });
+    pursePublishKit.publisher.publish(record);
+  };
+  // #endregion
+
+  /** @type { (desc: BrandDescriptor) => Promise<void>} */
+  const addIssuer = async desc => {
+    // assert haven't received this issuer before.
+    const descriptorsHas = brandDescriptors.has(desc.brand);
+    const pursesHas = brandPurses.has(desc.brand);
+    assert(
+      !(descriptorsHas && pursesHas),
+      'repeated brand from bank asset subscription',
+    );
+    assert(
+      !(descriptorsHas || pursesHas),
+      'corrupted state; one store has brand already',
+    );
+
+    // make a purse for this issuer.
+    // first make the purse so that if anything fails we don't update stores
+    const purse = await E(desc.issuer).makeEmptyPurse();
+
+    // save all five of these in a collection (indexed by brand?) so that when
+    // it's time to take an offer description you know where to get the
+    // relevant purse. when it's time to make an offer, you know how to make
+    // payments. REMEMBER when doing that, need to handle every exception to
+    // put the money back in the purse if anything fails.
+    brandDescriptors.init(desc.brand, desc);
+    brandPurses.init(desc.brand, purse);
+
+    publishPurses();
+  };
+
+  // Ensure a purse for each issuer
+  addIssuer({
+    brand: invitationIssuer.getBrand(),
+    issuer: invitationIssuer,
+    petname: 'invitations',
+  });
+  // watch the bank for new issuers to make purses out of
+  observeIteration(E(bank).getAssetSubscription(), {
+    async updateState(desc) {
+      await addIssuer({
+        brand: desc.brand,
+        issuer: desc.issuer,
+        petname: desc.proposedName,
+      });
+    },
+  });
+  /** @type {ERef<DepositFacet>} */
+  const depositFacet = Far('smart wallet deposit facet', {
+    /**
+     * Put the assets from the payment into the appropriate purse
+     *
+     * @param {Payment} payment
+     * @throws if the purse doesn't exist
+     * ??? PRODUCT: should it instead wait for an issuer/purse to put it in?
+     */
+    receive: payment => {
+      const brand = payment.getAllegedBrand();
+      const purse = brandPurses.get(brand);
+      const depositP = purse.deposit(payment);
+      E.when(depositP, publishPurses);
+      return depositP;
+    },
   });
 
-  const marshaller = wallet.getMarshaller();
+  const findInvitationInPurse = () => {
+    const invitationBrand = invitationIssuer.getBrand();
+    /** @type {Purse<'set'>} */
+    const purse = brandPurses.get(invitationBrand);
+    // FIXME find a matching one
+    const toWithdraw = purse.getCurrentAmount();
 
-  const myWalletStorageNode = E(storageNode).makeChildNode(address);
-  const storedSubscriber = makeStoredSubscriber(
-    subscriber,
-    myWalletStorageNode,
-    marshaller,
-  );
+    return purse.withdraw(toWithdraw);
+  };
+  const invitationFromSpec = makeInvitationsHelper(zoe, findInvitationInPurse);
 
+  const issuerSuggester = Far('issuerSuggester', {
```

Comment:
move this out of scope of the purses?

Reply:
It's this block,
```
    // ??? do we need this for PSM launch? if not better to remove the footgun to have assets that could get blown away in upgrade
    // can be held by the dapp. needed by the way we're doing AMM.
    // should have no possibility of unmarshalling a purse
    getIssuerSuggester: () => issuerSuggester,
```
so I'll just delete it for PS0.

Reply:
@samsiegart be advised... integration with the AMM (esp suggestIssuer) is likely to be delayed.

Reply:
Makes sense, thanks for the heads up